### PR TITLE
Cherry-pick #18683 to 7.x: LIBBEAT: Enhancement Convert dissected values from String to other basic data types and IP

### PR DESCRIPTION
**Pull request by sayden**
Cherry-pick of PR #18683 to 7.x branch. Original message: 

## What does this PR do?
This PR enhances dissect processor to convert string values to integer, long, float, double, boolean or IP. Key and convert data type are separated by `|` separator.

## Why is it important?
It will save CPU cycles converting data from string to other data type using Convert processor

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally
Added unit and benchmark test cases.

## Related issues

Closes https://github.com/elastic/dissect-specification/issues/10

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -384,6 +384,11 @@ field. You can revert this change by configuring tags for the module and omittin
 - Upgrade k8s.io/client-go and k8s keystore tests. {pull}18817[18817]
 - Add support for multiple sets of hints on autodiscover {pull}18883[18883]
 - Add a configurable delay between retries when an app metadata cannot be retrieved by `add_cloudfoundry_metadata`. {pull}19181[19181]
+- Add data type conversion in `dissect` processor for converting string values to other basic data types. {pull}18683[18683]
+- Add the `ignore_failure` configuration option to the dissect processor. {pull}19464[19464]
+- Add the `overwrite_keys` configuration option to the dissect processor. {pull}19464[19464]
+- Add support to trim captured values in the dissect processor. {pull}19464[19464]
+- Added the `max_cached_sessions` option to the script processor. {pull}19562[19562]
```
Comment (ycombinator) [inline CHANGELOG.next.asciidoc:391]: Extra entries :)

Comment (sayden) [reply CHANGELOG.next.asciidoc:391]: oh please, sorry

File: libbeat/processors/dissect/processor_test.go [hunk 2]
```diff
@@ -232,3 +232,179 @@ func TestErrorFlagging(t *testing.T) {
 		assert.Error(t, err)
 	})
 }
+
+func TestIgnoreFailure(t *testing.T) {
```
Comment (ycombinator) [inline libbeat/processors/dissect/processor_test.go:236]: These lines are coming from https://github.com/elastic/beats/pull/19685. Maybe we should get that merged first (ping @adriansr) so this backport PR here becomes cleaner?

Comment (kaiyan-sheng) [reply libbeat/processors/dissect/processor_test.go:236]: ahh sorry I missed this before approving...

Comment (sayden) [reply libbeat/processors/dissect/processor_test.go:236]: woaw! Good catch! You really impressed me :smiley:

Comment (ycombinator) [reply libbeat/processors/dissect/processor_test.go]: Hmmm, now it seems like we are completely missing [`TestProcessorConvert`](https://github.com/elastic/beats/pull/18683/files?file-filters%5B%5D=.go#diff-e47381c4db222aafe9fba3a0526cf71cR414) in this backport?

**Pull request merged**